### PR TITLE
Delete timesheet with Redux

### PR DESCRIPTION
**Pull request by tlorent**
Add the ability to delete an individual timesheet.

### CODE REVIEW

File: components/timesheets/Timesheets.jsx [hunk 1]
```diff
@@ -57,7 +57,7 @@ class Timesheets extends React.Component {
 Timesheets.propTypes = {
   requestTimesheets: PropTypes.func.isRequired,
   requestTimesheetsSuccess: PropTypes.func.isRequired,
-  timesheets: PropTypes.array.isRequired,
+  timesheets: PropTypes.arrayOf.isRequired,
```
Comment (boris-arkenaar) [inline components/timesheets/Timesheets.jsx]: `arrayOf` what?

File: ducks/timesheets.js [hunk 2]
```diff
@@ -26,7 +26,12 @@ export function timesheetsReducer(state = initialState, {
     case DEL_TIMESHEET:
       return { ...state, isLoading: true };
     case DEL_TIMESHEET_SUCCESS:
-      return { ...state, items: [...state.items.filter((timesheet) => timesheet.id !== id)], isLoading: false };
+      return {
+        ...state,
+        items: [...state.items.filter((timesheet) => timesheet.id
```
Comment (boris-arkenaar) [inline ducks/timesheets.js]: Je kopieert nu de volledige inhoud van een array, in een nieuwe array, zonder daar verder nog iets anders aan toe te voegen. Maw. de spread operator (en de wrapping array `[ ... ]`) is niet nodig; het resultaat van de `filter` functie is een array die je direct kan gebruiken.

File: pages/index.js [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
 import InfoBar from '../components/info-bar/InfoBar';
-import Timesheets from '../components/timesheets/Timesheets';
+import TimesheetsContainer from '../components/timesheets/TimesheetsContainer';
 
 const Index = () => (
   <React.Fragment>
     <Header />
     <InfoBar />
-    <Timesheets />
+    <TimesheetsContainer />
```
Comment (larstroost) [inline pages/index.js]: I thought we did not had to put the container in the index.js and we could still use Timesheets

File: ducks/timesheets.js [hunk 4]
```diff
@@ -19,6 +23,15 @@ export function timesheetsReducer(state = initialState, { timesheets, newTimeshe
       return { ...state, isLoading: true };
     case POST_TIMESHEET_SUCCESS:
       return { ...state, items: [...state.items, newTimesheet], isLoading: false };
+    case DEL_TIMESHEET:
+      return { ...state, isLoading: true };
+    case DEL_TIMESHEET_SUCCESS:
+      return {
+        ...state,
+        items: state.items.filter((timesheet) => timesheet.id
+        !== id),
```
Comment (boris-arkenaar) [inline ducks/timesheets.js]: I would place this on the previous line, for readability.

Comment (tlorent) [reply ducks/timesheets.js]: Thanks for the tip, I've placed it on the previous line now. When I still had it as ``items: [...state.items] etc`` the line was too long (>100 chars.) and object properties had to be put on their own line if they are not all on the same line (linter rules).

Comment (boris-arkenaar) [reply ducks/timesheets.js]: Yeah, I know. At least indent 'm in that case.

**Pull request merged**